languages/haskell/src/DMN/Translate/PY.hs

```diff
@@ -0,0 +1,146 @@
+annotationsAsComments chs dtrow =
+  let prefixedComments = catMaybes $ zipWith (\cheader commentcol -> ((varname cheader ++ ": ") ++) <$> commentcol) (comment_headers chs) (row_comments dtrow)
+      unprefixed = catMaybes $ row_comments dtrow
+  in
+  unlines $ ("    # "++) <$> (if length unprefixed > 1 then prefixedComments else unprefixed)
+
+fexp2js :: PYOpts -> ColHeader -> [FEELexp] -> String
+fexp2js jsopts ch fexps = wrapParen " or " (feel2pyIn ( showVarname jsopts ch) <$> fexps)
+
+showVarname :: PYOpts -> ColHeader -> String
+showVarname jsopts ch
+  | propstyle jsopts = "props[\"" ++ varname ch ++ "\"]"
+  | otherwise        = var_name ch
+               
+wrapParen :: String -> [String] -> String
+wrapParen myop xs
+  | length xs  > 1 = "(" ++ intercalate myop xs ++ ")"
+  | length xs == 1 = head xs
+  | otherwise      = "null"
+wrapArray :: String -> [String] -> String
+wrapArray myop xs = "[" ++ intercalate myop xs ++ "]"
+
+nonBlankCols :: a -> [FEELexp] -> Maybe (a, [FEELexp])
+nonBlankCols chs dtrows = if dtrows /= [FAnything] then Just (chs, dtrows) else Nothing
+
+input_headers :: [ColHeader] -> [ColHeader]
+input_headers   = filter ((DTCH_In==).label)
+
+comment_headers :: [ColHeader] -> [ColHeader]
+comment_headers = filter ((DTCH_Comment==).label)
+
+capitalize :: String -> String
+capitalize [] = []
+capitalize (x:xs) = toUpper x : xs
+
+feel2pyIn :: String -> FEELexp -> String
+feel2pyIn lhs  FAnything = wrapParen "or" ["True",lhs]
+feel2pyIn lhs (FSection Feq (VB rhs))  = lhs ++ "==" ++ capitalize (toLower <$> show rhs)
+feel2pyIn lhs (FSection Feq (VN rhs))  = lhs ++ "==" ++ show rhs
+feel2pyIn lhs (FSection Feq (VS rhs))  = lhs ++ "==" ++ show rhs
+feel2pyIn lhs (FSection Flt  (VN rhs)) = lhs ++ " < "  ++ show rhs
+feel2pyIn lhs (FSection Flte (VN rhs)) = lhs ++ " <="  ++ show rhs
+feel2pyIn lhs (FSection Fgt  (VN rhs)) = lhs ++ " > "  ++ show rhs
+feel2pyIn lhs (FSection Fgte (VN rhs)) = lhs ++ " >="  ++ show rhs
+feel2pyIn lhs (FInRange lower upper)   = wrapParen " and " [show lower ++ "<=" ++ lhs, lhs ++ "<=" ++ show upper]
+feel2pyIn lhs (FNullary rhs)           = feel2pyIn lhs (FSection Feq rhs)
+
+-- TODO:
+-- let's extend FEEL with support for PCRE lol
+
+-- if there's a single output column then we just return that value
+-- if there are multiple output columns then we construct an object with multiple properties and return the object.
+
+-- we could treat each output column as a lambda with access to the input namespace.
+-- if there's only one input column, then we honour sections by returning the boolean result of operating against the unnamed input
+-- if there are multiple input columns then we require explicit use of the input column varname
+
+feel2pyOut :: HitPolicy -> [ColHeader] -> DTrow -> (Maybe String, [String])
+feel2pyOut hp chs dtrow
+  -- ("// one input column, allowing binary operators in output column(s)"
+  = (Nothing, -- "toreturn[thiscolumn] = ..."
+     uncurry showFeels <$> zip
```

Comment:
suggest refactoring so that the lambda expressions output correctly for both the JS module and the PY module. You could add an argument to `showFeel` which produces the correct syntax for a lambda depending on JS vs PY; and the caller (JS vs PY) could provide that syntax when it calls `showFeel`.